lib/internal/test_runner/utils.js

```diff
@@ -222,6 +224,48 @@ function parseCommandLine() {
   return globalTestOptions;
 }
 
+function coverageThreshold(coverage, color) {
+  coverage = NumberPrototypeToFixed(coverage, 2);
+  if (color) {
+    if (coverage > 90) return `${green}${coverage}${color}`;
+    if (coverage < 50) return `${red}${coverage}${color}`;
+  }
+  return coverage;
+}
+
+function getCoverageReport(pad, summary, symbol, color) {
+  let report = `${color}${pad}${symbol}start of coverage report\n`;
+
+  report += `${pad}${symbol}file | line % | branch % | funcs % | uncovered lines\n`;
+
+  for (let i = 0; i < summary.files.length; ++i) {
+    const {
+      path,
+      coveredLinePercent,
+      coveredBranchPercent,
+      coveredFunctionPercent,
+      uncoveredLineNumbers,
+    } = summary.files[i];
+    const relativePath = relative(summary.workingDirectory, path);
+    const lines = coverageThreshold(coveredLinePercent, color);
+    const branches = coverageThreshold(coveredBranchPercent, color);
+    const functions = coverageThreshold(coveredFunctionPercent, color);
+    const uncovered = ArrayPrototypeJoin(uncoveredLineNumbers, ', ');
+
+    report += `${pad}${symbol}${relativePath} | ${lines} | ${branches} | ` +
+              `${functions} | ${uncovered}\n`;
+  }
+
+  const { totals } = summary;
+  report += `${pad}${symbol}all files | ` +
+            `${coverageThreshold(totals.coveredLinePercent, color)} | ` +
+            `${coverageThreshold(totals.coveredBranchPercent, color)} | ` +
+            `${coverageThreshold(totals.coveredFunctionPercent, color)} |\n`;
+
+  report += `${pad}${symbol}end of coverage report${white}\n`;
```

Comment:
Should `white` be used here? I'm trying to avoid generating colors for the TAP reporter. If this is to reset the output color, then maybe do something like:

```js
if (color) {
  report += white
}
```